corehq/apps/email/models.py

```diff
@@ -1,13 +1,19 @@
 from django.db import models
 
 from corehq.motech.const import PASSWORD_PLACEHOLDER, ALGO_AES
-from corehq.motech.utils import b64_aes_decrypt, b64_aes_encrypt
+from corehq.motech.utils import (
+    b64_aes_cbc_decrypt,
+    b64_aes_cbc_encrypt,
+    b64_aes_decrypt,
+    b64_aes_encrypt,
+)
 
 
 class EmailSettings(models.Model):
     domain = models.CharField(max_length=255, unique=True)
     username = models.CharField(max_length=255)
-    password = models.CharField(max_length=255)
+    password = models.CharField(max_length=255)  # This will be replaced by password_cbc after migration is done
+    password_cbc = models.CharField(max_length=255)
```

Comment:
Instead of creating a new field, just use a different prefix.

Define a new prefix, say,
```python
# corehq/motech/const.py
ALGO_AES_CBC = 'aes-cbc'
```

And then, you can use it to encrypt and decrypt:
```python
    @property
    def plaintext_password(self):
        if self.password.startswith(f'${ALGO_AES_CBC}$'):
            ciphertext = self.password.split('$', 2)[2]
            return b64_aes_cbc_decrypt(ciphertext)
        if self.password.startswith(f'${ALGO_AES}$'):
            ciphertext = self.password.split('$', 2)[2]
            return b64_aes_decrypt(ciphertext)
        return self.password

    @plaintext_password.setter
    def plaintext_password(self, plaintext):
        if plaintext != PASSWORD_PLACEHOLDER:
            ciphertext = b64_aes_cbc_encrypt(plaintext)
            self.password = f'${ALGO_AES_CBC}${ciphertext}'
```

Reply:
+1

Reply:
Thanks for the suggestion - this way is cleaner. [ac20dac](https://github.com/dimagi/commcare-hq/pull/35403/commits/ac20dacdf690e6e61f6d7f0a5c386c6dc4d54e08)

Jing and I did consider this but decided to create another field so that we're not overriding existing data just to make the transition a bit safer. But looking at it again, it's pretty straightforward so i'm comfortable doing it this way. 

Just curious - is the main benefit of this strategy to reduce overhead of creating a new field and eventually having to delete the original? Or is there anything else additional?

Reply:
I think the main benefit is simplicity of implementation and rollout. It's much easier to roll out if there are no database columns to be added/removed.